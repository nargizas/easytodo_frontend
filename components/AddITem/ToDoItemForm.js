/* eslint-disable react/jsx-no-bind */
/* eslint-disable react/prop-types */
import { React, useState } from 'react';
import {
  View, StyleSheet, Alert, Platform,
} from 'react-native';
import Input from './Input';
import DateTimeInput from './DateTimeInput';
import PrimaryButton from '../PrimaryButton';

const styles = StyleSheet.create({
  appContainer: {
    flex: 1,
  },
  inputContainer: {
    flex: 2,
  },
  buttonsContainer: {
    justifyContent: 'flex-end',
    // alignItems: 'flex-end',
    alignContent: 'flex-end',
    flex: 1,
    marginBottom: 8,
  },
});

function ToDoItemForm({
  suggestedTitle, cancelHandler, confirmHandler, submitButtonLabel, defaultValues,
}) {
  const [inputValues, setInputValues] = useState({
    title: defaultValues ? defaultValues.title : suggestedTitle || '',
    deadline: defaultValues ? defaultValues.deadline : new Date(),
    status: 'IN_PROGRESS',
  });

  function inputChangeHandler(inputIdentifier, enteredValue) {
    setInputValues((currentInputValues) => ({
      ...currentInputValues,
      [inputIdentifier]: enteredValue,
    }));
  }

  const [date, setDate] = useState(inputValues.deadline);
  const [mode, setMode] = useState('date');
  const [show, setShow] = useState(false);

  const onChange = (event, selectedDate) => {
    const currentDate = selectedDate;
    setShow(false);
    setDate(currentDate);
    setInputValues((currentInputValues) => ({
      ...currentInputValues,
      deadline: currentDate,
    }));
  };

  const showMode = (currentMode) => {
    if (Platform.OS === 'android') {
      setShow(true);
      // for iOS, add a button that closes the picker
    }
    setMode(currentMode);
  };

  const showDatepicker = () => {
    showMode('date');
  };

  const showTimepicker = () => {
    showMode('time');
  };

  function submitHandler() {
    const toDoItemData = {
      title: inputValues.title,
      deadline: new Date(inputValues.deadline),
      status: inputValues.status,
    };

    const titleIsValid = toDoItemData.title.trim().length > 0;
    if (!titleIsValid) {
      Alert.alert('Invalid input', 'Please check your title');
      return;
    }
    confirmHandler(toDoItemData);
  }

  return (
    <View style={styles.appContainer}>
      <View style={styles.inputContainer}>
        <Input
          label="Title"
          textInputConfig={{
            onChangeText: inputChangeHandler.bind(this, 'title'),
            value: inputValues.title,
          }}
        />
        <DateTimeInput
          date={date}
          show={show}
          dateInputConfig={{
            onChange,
            testID: 'dateTimePicker',
            value: date,
            mode,
            is24Hour: true,
          }}
          showDatepicker={showDatepicker}
          showTimepicker={showTimepicker}
        />
      </View>

      <View style={styles.buttonsContainer}>
        <PrimaryButton onPress={cancelHandler}>Cancel</PrimaryButton>
        <PrimaryButton onPress={submitHandler}>{submitButtonLabel}</PrimaryButton>
      </View>
    </View>
  );
}

export default ToDoItemForm;
