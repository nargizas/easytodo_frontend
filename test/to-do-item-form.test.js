import React from 'react';
import renderer from 'react-test-renderer';

import ToDoItemForm from '../components/AddITem/ToDoItemForm';

it('renders correctly', () => {
  const tree = renderer.create(<ToDoItemForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
