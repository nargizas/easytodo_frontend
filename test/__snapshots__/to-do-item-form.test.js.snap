// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 2,
      }
    }
  >
    <View
      style={
        {
          "marginHorizontal": 16,
          "marginVertical": 8,
        }
      }
    >
      <Text
        style={
          {
            "color": "#f5f5f5",
            "fontSize": 18,
            "marginBottom": 4,
          }
        }
      >
        Title
      </Text>
      <TextInput
        onChangeText={[Function]}
        style={
          {
            "backgroundColor": "#f5f5f5",
            "borderRadius": 8,
            "elevation": 2,
            "fontSize": 18,
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          }
        }
        value=""
      />
    </View>
    <View
      style={
        {
          "marginHorizontal": 16,
          "marginVertical": 8,
        }
      }
    >
      <Text
        style={
          {
            "color": "#f5f5f5",
            "fontSize": 18,
            "marginBottom": 4,
          }
        }
      >
        Deadline
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flex": 1,
            }
          }
        >
          <View
            style={
              {
                "alignContent": "center",
                "alignItems": "center",
                "backgroundColor": "#f5f5f5",
                "borderRadius": 16,
                "justifyContent": "center",
                "marginHorizontal": 16,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 16,
                }
              }
            >
              Pick date
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flex": 1,
            }
          }
        >
          <View
            style={
              {
                "alignContent": "center",
                "alignItems": "center",
                "backgroundColor": "#f5f5f5",
                "borderRadius": 16,
                "justifyContent": "center",
                "marginHorizontal": 16,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 16,
                }
              }
            >
              Pick time
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <Text
          style={
            [
              {
                "marginHorizontal": 16,
                "marginVertical": 8,
              },
              {
                "fontSize": 16,
              },
              {
                "color": "#f5f5f5",
              },
            ]
          }
        >
          Selected:
           
          2022-12-17 14:38
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignContent": "flex-end",
        "flex": 1,
        "justifyContent": "flex-end",
        "marginBottom": 8,
      }
    }
  >
    <View
      style={
        {
          "borderRadius": 36,
          "marginHorizontal": 16,
          "marginVertical": 8,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#f5f5f5",
            "elevation": 2,
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "black",
              "fontSize": 18,
              "opacity": 54,
              "textAlign": "center",
            }
          }
        >
          Cancel
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "borderRadius": 36,
          "marginHorizontal": 16,
          "marginVertical": 8,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#f5f5f5",
            "elevation": 2,
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "black",
              "fontSize": 18,
              "opacity": 54,
              "textAlign": "center",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
